xbmc/cores/VideoPlayer/DVDCodecs/DVDFactoryCodec.cpp

```diff
@@ -246,45 +250,29 @@ CDVDOverlayCodec* CDVDFactoryCodec::CreateOverlayCodec( CDVDStreamInfo &hint )
   {
     case AV_CODEC_ID_TEXT:
     case AV_CODEC_ID_SUBRIP:
-      pCodec.reset(new CDVDOverlayCodecText());
-      if (pCodec->Open(hint, options))
-      {
-        return pCodec.release();
-      }
+      pCodec = std::make_unique<CDVDOverlayCodecText>();
       break;
 
     case AV_CODEC_ID_SSA:
     case AV_CODEC_ID_ASS:
-      pCodec.reset(new CDVDOverlayCodecSSA());
-      if (pCodec->Open(hint, options))
-      {
-        return pCodec.release();
-      }
-
-      pCodec.reset(new CDVDOverlayCodecText());
-      if (pCodec->Open(hint, options))
-      {
-        return pCodec.release();
-      }
+      pCodec = std::make_unique<CDVDOverlayCodecSSA>();
       break;
 
+      //! @todo: this seems unused?
+      pCodec = std::make_unique<CDVDOverlayCodecText>();
```

Comment:
@enen92 What do you think about this? Should I just drop it from here as it seems it will never be used

Reply:
Yeah please drop it if possible, the same for the ASS condition in the actual codec

Reply:
updated, thanks!